src/videojs-hls.js

```diff
@@ -506,11 +506,23 @@ videojs.HlsHandler.prototype.setCurrentTime = function(currentTime) {
 };
 
 videojs.HlsHandler.prototype.duration = function() {
-  var playlists = this.playlists;
+  var
+    playlists = this.playlists,
+    playlistDuration;
+
   if (playlists) {
-    return videojs.Hls.Playlist.duration(playlists.media());
+    playlistDuration = videojs.Hls.Playlist.duration(playlists.media());
+  } else {
+    return 0;
+  }
+
+  if (playlistDuration === Infinity) {
+    return playlistDuration;
+  } else if (this.mediaSource) {
+    return this.mediaSource.duration;
```

Comment:
Because of the way we fake mediaSource duration during live playlists, we need to return the playlist's duration because the mediaSource.duration will be an actual value so we can seek to the live point on play.